[Doc] edit doc for survival_difference_at_fixed_point_in_time_test

DanBasson opened the pull request:
related issue - #1446 

Elaborate on survival_difference_at_fixed_point_in_time_test:

- added image for survival plot

- added the ability to plot point in time test in plot_survival_function

- added output for results


note that the formatting changes in plottng.py is taken from the black playground

CamDavidsonPilon commented:
This is great! Thanks @DanBasson! 

Are the changes to plotting.py relevant or just lint changes?

DanBasson commented:
I've also added the option to insert "point_in_time" to the "plot_survival_function".

I've added the following (lines 955-957):

```
if "point_in_time" in kwargs:  # marker for given point
    point_in_time = kwargs["point_in_time"]
    kwargs.pop("point_in_time")
```

and (lines 1024-1027):

```
if "point_in_time" in locals():
        plot_estimate_config.ax.scatter(
            point_in_time, cls.survival_function_at_times(point_in_time)
        )
```

CamDavidsonPilon commented:
👍 cool lgtm
